app/controllers/concerns/edge_cache_safety_check.rb

```diff
@@ -0,0 +1,17 @@
+module EdgeCacheSafetyCheck
+  extend ActiveSupport::Concern
+
+  CANNOT_USE_CURRENT_USER = "You may not use current_user in this cached code path.".freeze
+
+  def current_user
+    # In production, using current_user in a view which will be cached will result in a cache leak.
+    # https://docs.forem.com/technical-overview/architecture/#we-cache-many-content-pages-on-the-edge
+    return super unless @edge_caching_in_place
+
+    return if session_current_user_id.blank?
+
+    @current_user_called_erroneously = true
+
+    CANNOT_USE_CURRENT_USER
```

Comment:
Shouldn't this raise, not just return a string instead of a user object?

Reply:
The issue is that `user_signed_in?` still uses `current_user` for truthiness. So returning the string means that it will show the error if someone tries to call a method on `current_user` but `user_signed_in?` and anything else that relies on `current_user` for boolean presence will still work.